pep-9999.rst

```diff
@@ -180,6 +180,17 @@ Yes::
         def do_stuff(self, y: str) -> None: ...
         doStuff = do_stuff
 
+The type of generic classes can be narrowed by annotating the ``self``
+argument of the ``__init__`` method::
+
+    class Foo(Generic[_S, _T]):
+        @overload
+        def __init__(self: Foo[str, str], type: Literal["s"]) -> None: ...
+        @overload
+        def __init__(self: Foo[int, _T], type: Literal["i"]) -> None: ...
```

Comment:
How would `_T` be determined in this example? Maybe better to use a simpler example with just one TypeVar.

Reply:
The proposed change that initiated this documentation request has two type variables, so it would probably be good to include a two-variable example. But @JelleZijlstra makes a good point that `_T` should appear elsewhere in the signature so it can be solved by the TypeVar constraint engine.